### DISCUSSION TITLE
NGINX 504 Timeout (Optimize / Stats / etc.)

### DISCUSSION THREAD

**Discussion by quella** [FreshRSS/FreshRSS#3557] [Q&A]
I'm currently running FreshRSS 1.18.1-Dev on a local RaspberryPI 3B+.  I am beginning to see more Error 504 Timeout messages from the Web interface when I attempt to run the Optimize Database function and attempt to view the Stats page.  In version 1.17 I would get this 504 error when doing the Optimize function which would require a refresh when it was done.  Yet, getting the same error when trying the Stats page after updating to 1.18.1-dev.

Are there timeout setting for NGINX that I could modify limiting these errors or other suggestions someone can recommend?

![Screen Shot 2021-03-23 at 10 22 05 AM](https://user-images.githubusercontent.com/823352/112163170-26287000-8bc3-11eb-8b38-da6cc06beb03.png)

![Screen Shot 2021-03-23 at 10 22 22 AM](https://user-images.githubusercontent.com/823352/112163190-2aed2400-8bc3-11eb-891f-54839580ac33.png)

![Screen Shot 2021-03-23 at 10 22 43 AM](https://user-images.githubusercontent.com/823352/112163207-2f194180-8bc3-11eb-9b65-c89a0eeeb1ea.png)

Quella

**Comment by Alkarex** [comment]
I do not recommend nginx, but see the several timeout related parameters, e.g.
https://www.nginx.com/resources/wiki/start/topics/examples/full/#proxy-conf

**Reply by quella** [reply]
Thanks for the link and I believe I followed a guide on installing FreshRSS on a Raspberry Pi and it was my mistake to think this was a requirement for the application.  I did find the following setting for NGINX (https://easycloudsupport.zendesk.com/hc/en-us/articles/360002057472-How-to-Fix-504-Gateway-Timeout-using-Nginx) and I modified the "max_execution_timeout" from 30 seconds (default) to 60 and the stats page comes back as expected.  Thanks for the pointers and helping me understand that this error was not a part of the FreshRSS tool.
